compiler/rustc_const_eval/src/interpret/machine.rs

```diff
@@ -285,20 +285,33 @@ pub trait Machine<'mir, 'tcx>: Sized {
         ptr: Pointer,
     ) -> Pointer<Self::PointerTag>;
 
-    /// "Int-to-pointer cast"
+    /// Hook for returning a pointer from a transmute-like operation on an addr.
     fn ptr_from_addr(
```

Comment:
Maybe call this "ptr_from_addr_transmute" and the other one "ptr_from_addr_cast"?